runtime/vm/JVMImage.cpp

```diff
@@ -451,33 +515,45 @@ registerCPEntry(J9JavaVM *javaVM, J9ClassPathEntry *cpEntry)
 }
 
 extern "C" void
-deregisterClassLoader(J9JavaVM *javaVM, J9ClassLoader *classLoader)
+deregisterClass(J9JavaVM *javaVM, J9Class *clazz)
 {
 	IMAGE_ACCESS_FROM_JAVAVM(javaVM);
 
 	Assert_VM_notNull(jvmImage);
 
-	jvmImage->deregisterEntryInTable(jvmImage->getClassLoaderTable(), (UDATA)classLoader);
+	jvmImage->deregisterEntryInTable(jvmImage->getClassTable(), (UDATA)clazz);
 }
 
 extern "C" void
-deregisterClass(J9JavaVM *javaVM, J9Class *clazz)
+deregisterCPEntry(J9JavaVM *javaVM, J9ClassPathEntry *cpEntry)
 {
 	IMAGE_ACCESS_FROM_JAVAVM(javaVM);
 
 	Assert_VM_notNull(jvmImage);
 
-	jvmImage->deregisterEntryInTable(jvmImage->getClassTable(), (UDATA)clazz);
+	jvmImage->deregisterEntryInTable(jvmImage->getClassPathEntryTable(), (UDATA)cpEntry);
 }
 
-extern "C" void
-deregisterCPEntry(J9JavaVM *javaVM, J9ClassPathEntry *cpEntry)
+extern "C" J9ClassLoader *
+findClassLoader(J9JavaVM *javaVM, uint32_t classLoaderCategory)
 {
 	IMAGE_ACCESS_FROM_JAVAVM(javaVM);
 
-	Assert_VM_notNull(jvmImage);
+	/* TODO: Function will change when hash table is created and there would be no need for accessing specific class loaders (hardcoded) */
```

Comment:
What if we just did a switch statement here instead. With the default case being return NULL?
Minor nitpick

Reply:
this will be changed in the future for a hash table